### PR TITLE
Add TypeScript definition

### PR DESCRIPTION
**Pull request by BendingBender**

### CODE REVIEW

File: index.test-d.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import logUpdate, {stderr, create} from '.';
```
Comment (sindresorhus) [inline index.test-d.ts]: The named exports really bother me. They come with totally generic and useless names, which means everyone would have to rename them to `logUpdateStderr` or something. Why can't we just add them as properties on the default export?

File: index.js [hunk 2]
```diff
@@ -58,5 +58,6 @@ const main = (stream, options) => {
 };
 
 module.exports = main(process.stdout);
+module.exports.default = main(process.stdout);
```
Comment (sindresorhus) [inline index.js]:
```suggestion
module.exports.default = module.export
```

For consistency and to ensure they're the same instance.

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	 *     showCursor: true
+	 * });
+	 */
+	readonly showCursor?: boolean;
+}
+
+/**
+ * Get a `logUpdate` method that logs to the specified stream.
+ *
+ * @param stream The stream to log to.
```
Comment (sindresorhus) [inline index.d.ts]:
```suggestion
 * @param stream - The stream to log to.
```

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+export interface Options {
+	/**
+	 * Show the cursor. This can be useful when a CLI accepts input from a user.
+	 *
+	 * @example
+	 *
+	 * // Write output but don't hide the cursor
+	 * const log = logUpdate.create(process.stdout, {
+	 *     showCursor: true
```
Comment (sindresorhus) [inline index.d.ts]: This should be tab-indented:

```suggestion
	 * 	showCursor: true
```

File: index.test-d.ts (hunk 1 above)
Comment (BendingBender) [reply index.test-d.ts]: It's doable. do you want me to change that?

File: index.js (hunk 2 above)
Comment (BendingBender) [reply index.js]: Oh, right, overlooked this one.

File: index.test-d.ts (hunk 1 above)
Comment (BendingBender) [reply index.test-d.ts]: Done.

Comment (sindresorhus) [reply index.test-d.ts]: You have to update this import statement too then.

Comment (BendingBender) [reply index.test-d.ts]: I'm testing both kinds of imports.

Comment (sindresorhus) [reply index.test-d.ts]: I don't want to support two different ways of using my modules.

**Comment by sindresorhus** [comment]
Relevant: https://twitter.com/sindresorhus/status/1102856012103475200

**Pull request merged**